docs/guides/docs/intro-to-sway/contract-functions.mdx

```diff
@@ -0,0 +1,315 @@
+
+<CodeImport
+  file="../../../intro-to-sway/contract/src/main.sw"
+  comment="buy_item_get_item"
+  commentType="//"
+  lang="sway"
+/>
+
+In Sway, all variables are immutable by default, whether declared with `let` or `const`. To modify the value of any variable, it must be declared mutable using the `mut` keyword. Since we plan to update the item's `total_bought` value, it should be defined as mutable.
+
+Additionally, it's essential to ensure that the quantity of coins sent for the item isn't less than the item's price.
+
+<CodeImport
+  file="../../../intro-to-sway/contract/src/main.sw"
+  comment="buy_item_require_ge_amount"
+  commentType="//"
+  lang="sway"
+/>
+
+### Updating storage
+
+We can increase the item's `total_bought` field value and subsequently reinsert it into the `item_map`. This action will replace the earlier value with the revised item.
+
+<CodeImport
+  file="../../../intro-to-sway/contract/src/main.sw"
+  comment="buy_item_require_update_storage"
+  commentType="//"
+  lang="sway"
+/>
+
+### Transferring payment
+
+Lastly, we can process the payment to the seller. It's recommended to transfer assets only after all storage modifications are completed to prevent [re-entrancy attacks](https://fuellabs.github.io/sway/v0.32.1/book/blockchain-development/calling_contracts.html#handling-re-entrancy).
+
+For items reaching a specific price threshold, a fee can be deducted using a conditional `if` statement. The structure of `if` statements in Sway mirrors that in JavaScript.
+
+<CodeImport
+  file="../../../intro-to-sway/contract/src/main.sw"
+  comment="buy_item_require_transferring_payment"
+  commentType="//"
+  lang="sway"
+/>
+
+In the aforementioned if-condition, we assess if the transmitted amount surpasses 100,000,000. For clarity in large numbers like `100000000`, we can represent it as `100_000_000`. If the foundational asset for this contract is ETH, this equates to 0.1 ETH given that Fuel employs a 9 decimal system.
+
+Should the amount exceed 0.1 ETH, a commission is determined and then deducted from the total.
+
+To facilitate the payment to the item's owner, the `transfer` function is utilized. This function, sourced from the standard library, requires three parameters: the coin quantity for transfer, the coin's asset ID, and the Identity to which the coins are dispatched.
+
+## 3. Get an item
+
+To get the details for an item, we can create a read-only function that returns the `Item` struct for a given item ID.
+
+<CodeImport
+  file="../../../intro-to-sway/contract/src/main.sw"
+  comment="get_item"
+  commentType="//"
+  lang="sway"
+/>
+
+
+To return a value in a function, you can use the `return` keyword, similar to JavaScript. Alternatively, you can omit the semicolon in the last line to return that value. Although both methods are effective, it's generally better to be explicit.
```

Comment:
I will remove "it's generally better to be explicit." but I'm going to leave returns in the examples because this detail can be easily overlooked if you are coming from Javascript and not familar with rust.